patternbuilder.install

```diff
@@ -79,6 +79,267 @@ function patternbuilder_modules_enabled($modules) {
+  );
+  // 'columns' are hardcoded in the field type.
+  $field['columns'] = $field_schema['columns'];
+  // 'foreign keys' are hardcoded in the field type.
+  $field['foreign keys'] = $field_schema['foreign keys'];
+  // 'indexes' can be both hardcoded in the field type, and specified in the
+  // incoming $field definition.
+  $field += array(
+    'indexes' => array(),
+  );
+  $field['indexes'] += $field_schema['indexes'];
+
+  // Get the storage schema in case it's differnt than the field schema.
+  $storage_schema = _field_sql_storage_schema($field);
+  if (empty($storage_schema)) {
+    return FALSE;
+  }
+
+  $have_transactional_support = Database::getConnection()->supportsTransactionalDDL();
+  if ($have_transactional_support) {
+    // If the database supports transactional DDL, we can go ahead and rely
+    // on it. If not, we will have to rollback manually if something fails.
+    $transaction = db_transaction();
+  }
+
+  $converted = FALSE;
+  try {
+    // Update the field storage tables.
+    // Text fields all have the same indexes so this only needs to update the
+    // field schema columns.
+    // Note: This only updates or adds columns. It does not remove columns.
+    // If converting to a type that has fewer columns, then the original
+    // columns will remain.
+    foreach ($storage_schema as $table_name => $table_schema) {
+      foreach ($field['columns'] as $field_column_name => $field_attributes) {
+        $column_name = _field_sql_storage_columnname($field['field_name'], $field_column_name);
+        if (isset($table_schema['fields'][$column_name])) {
+          if (db_field_exists($table_name, $column_name)) {
+            db_change_field($table_name, $column_name, $column_name, $table_schema['fields'][$column_name]);
+          }
+          else {
+            db_add_field($table_name, $column_name, $table_schema['fields'][$column_name]);
+          }
+        }
+      }
+    }
+
+    // Tell the storage engine to update the field. Do this before
+    // saving the new definition since it still might fail.
+    $storage_type = field_info_storage_types($field['storage']['type']);
+    module_invoke($storage_type['module'], 'field_storage_update_field', $field, $field_original, $has_data);
+
+    // Update the field config.
+    // The serialized 'data' column contains everything from $field that does not
+    // have its own column and is not automatically populated when the field is
+    // read.
+    $data = $field;
+    unset($data['columns'], $data['field_name'], $data['type'], $data['locked'], $data['module'], $data['cardinality'], $data['active'], $data['deleted']);
+    // Additionally, do not save the 'bundles' property populated by
+    // field_info_field().
+    unset($data['bundles']);
+
+    $field['data'] = $data;
+
+    $field_config_primary_key = array('id');
+    drupal_write_record('field_config', $field, $field_config_primary_key);
+
+    // Clear caches
+    field_cache_clear(TRUE);
```

Comment:
I'll update it.. the odd thing is that is how field_update_field calls it too - https://api.drupal.org/api/drupal/modules%21field%21field.crud.inc/function/field_update_field/7.x